components/gantry/oneaxis/oneaxis.go

```diff
@@ -187,10 +189,79 @@ func (g *oneAxis) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 			return err
 		}
 	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	g.cancelFunc = cancelFunc
```

Comment:
What happened to the old value of `g.cancelFunc`? If we never call it, won't reconfiguring the component leak background coroutines that never shut down?

I suspect the way to fix this is that at the top of `Reconfigure` we put something like this:
```go
if g.cancelFunc != nil {
    g.cancelFunc()
    g.activeBackgroundWorkers.Wait()
}
```
and now we can overwrite these variables and call `g.checkHit` later, without worrying about this stuff already running from a previous reconfiguration.

...and then given my worries in `Close()`, we should also make sure that the motor is not moving during reconfiguration.

Reply:
so just stop the motor at the beginning of reconfigure?

Reply:
@penguinland do you mean why you'd want to change the limit switch pins specifically?
Upgrading hardware is a thing.
Messing up which pin is connected to which in a large gantry you're prototyping is a thing.
Discovering you've connected to an spi pin you want to use on a board and having to change up the pins is a thing.

I'd rather stop the motor, I think, and we already stop our own motor on (re)creation and (re)configuration. I think the Stop -> stop context -> start context is safer. Why would you want to continue the operation and a coroutine through reconfiguration?

Reply:
I meant is there a reason to replace the old coroutine and cancelFunc with new ones. I agree that you might want to replace the hardware. I'm perfectly content replacing them as long as we shut down the old coroutine and cancelFunc first.